### PR TITLE
feat: Get Invitations endpoint

### PR DESCRIPTION
**Pull request by olugbenga1**
## What does the PR do?
This PR is about getting an organisation's invitations

## How should this be manually tested?
Request Example 

    GET /api/v1/organisations/invitations

#### Success Response 

```
{
	"status_code": 200
	"message": "Request successful",
	"data": [
	{
		"email": "[user@example.com](mailto:user@example.com)",
		"org_id": "aoalj-alhljkdshj345klhsd",
		"status: "pending/approved/rejected",
                "sent_at": "2023-06-23T00:00:00Z",
                "expires_at": "2023-07-23T00:00:00Z"
	}
      ]
}
```

#### Failure Response
```
{
	"error": "Invalid request",
	"message": "User not authenticated or authorized"
}
```

### Task
- [x] Implement authentication logic to ensure the client is signed in
- [x] Implement authorization logic to verify the client is an admin.
- [x] Retrieve all user invitations sent by the admin.
- [x] Ensure the endpoint correctly identifies if the client is signed in
- [x] Confirm the endpoint verifies that the client is an admin
- [x] Test the endpoint correctly fetches all invitations sent by the admin.

## Reference Issue
[[FEAT]: Get organisation invitations](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/172)

### CODE REVIEW

**Review by Homoakin619 — APPROVED** [review]

**Comment by Homoakin619** [comment]
Your build is failing. Fix your errors

**Review by TheCodeGhinux — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

File: src/database/seeding/seeding.service.ts
Comment (buka4rill) [inline src/database/seeding/seeding.service.ts:1]: Linting wrong on this file

Comment (olugbenga1) [reply src/database/seeding/seeding.service.ts:1]: Thanks for the feedback. Could you provide more details on the specific linting rules being violated? I'm unable to identify the error and pulled that file from the remote branch.

**Comment by TheCodeGhinux** [comment]
@buka4rill please look into this

File: src/modules/invite/dto/create-invite.dto.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export class CreateInviteDto {}
```
Comment (buka4rill) [inline src/modules/invite/dto/create-invite.dto.ts:1]: Why is this empty?

File: src/modules/invite/invite.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  update(id: number, updateInviteDto: UpdateInviteDto) {
+    return `This action updates a #${id} invite`;
+  }
+
+  remove(id: number) {
+    return `This action removes a #${id} invite`;
+  }
+}
```
Comment (buka4rill) [inline src/modules/invite/invite.service.ts:48]: If these are to be implemented later add a TODO comment
`// TODO: to be implemented`

**Review by buka4rill — COMMENTED** [review]
left a few comments

File: src/modules/invite/dto/create-invite.dto.ts (hunk 1 above)
Comment (olugbenga1) [reply src/modules/invite/dto/create-invite.dto.ts]: I have now removed this file and implemented invite dto

File: src/modules/invite/invite.service.ts (hunk 2 above)
Comment (olugbenga1) [reply src/modules/invite/invite.service.ts]: I have fixed this

**Review by buka4rill — APPROVED** [review]

**Pull request merged**